nats.go

```diff
@@ -5455,6 +5456,66 @@ func (nc *Conn) GetClientID() (uint64, error) {
 	return nc.info.CID, nil
 }
 
+// ConnStatusChanged returns a channel on which connection status changes will be reported.
+// Reported statuses: CONNECTED, RECONNECTING, DISCONNECTED, CLOSED.
+func (nc *Conn) ConnStatusChanged() chan Status {
```

Comment:
How about:

```go
func(nc *Conn) StatusChanged(...Status) chan Status 
```

so that it lines up better with the getter for connection status:
```go
func(nc *Conn) Status() Status
```

also ideally this would be a receive only channel here when but closing the channel seems the most straightforward way to unregister these.

Reply:
Yes, I like this.  Might be some value in being able to supply own chan but happy to leave that till a real need arise.

Reply:
Agreed, I also like that. I was thinking more granular approach with providing own channel per-status might be useful, but I agree that maybe this is something we can add in the future if necessary.

And yeah, receive-only channel would be ideal and that's what I wanted to to do at first, but closing the channel seemed like a good way to unregister.